Adrien/feature/added twitter to reddit

why-not-try-calmer opened the pull request:
Continuation of #29. 

To do:
- add tests
- add documentation

KaratekHD committed:
Implement basic caching

why-not-try-calmer committed:
- added sync_utils.py

KaratekHD committed:
Add .env to .gitignore

why-not-try-calmer committed:
---

why-not-try-calmer committed:
reset

why-not-try-calmer committed:
cleaned up tests

why-not-try-calmer committed:
Cache MiddleWare with in-line documentation and tests

why-not-try-calmer committed:
- cleanedup

why-not-try-calmer committed:
much better design

why-not-try-calmer committed:
- refactored entirely
- endpoints tests are missing

why-not-try-calmer committed:
cleaned up

why-not-try-calmer committed:
- changed initialization to a "modules subscribe themselves" pattern

why-not-try-calmer committed:
more cleanups, more flexibility

why-not-try-calmer committed:
finished for tonight

KaratekHD commented on defrag/__init__.py:
```diff
@@ -33,5 +34,17 @@
 LOAD = []
 
+
+def pretty_log(comment: str, value: Optional[str] = None, warn: bool = False) -> None:
```
Why do we need this? Isn't `LOGGER` enough?

why-not-try-calmer replied:
Have you tried to see the difference?

KaratekHD commented on defrag/modules/twitter.py:
```diff
@@ -0,0 +1,66 @@
+            print("Unable to fetch from Twitter @openSUSE: ", err)
+
+
+def register_service():
```
I am a bit confused, where do you call this?

why-not-try-calmer replied:
Wherever we want! Check out the unittests for an example, the new documentation under this function from my latest commit for more details, and the docs of the ServiceManager class for the bigger picture.

why-not-try-calmer committed:
- added (successful on my machines) tests

why-not-try-calmer committed:
- added documentation to all new classes
- added tests

why-not-try-calmer committed:
- added refresh instance method
- added global runner for refreshing service stores
- added global runner verification when registering a service
- added modules-as-services registration

why-not-try-calmer committed:
.

why-not-try-calmer committed:
last cleanup for the day

why-not-try-calmer commented:
Okay, I think this PR is now worth considering for a merge. But before that, maybe you have a clue why the unittests for reddit and twitter fail (404) when I get exactly the correct result when querying the app locally from a web browser?

why-not-try-calmer committed:
- added lock

why-not-try-calmer committed:
- edited documentation

why-not-try-calmer committed:
- made stores accessors async
- cleaned up

KaratekHD commented:
> Okay, I think this PR is now worth considering for a merge. But before that, maybe you have a clue why the unittests for reddit and twitter fail (404) even though I am getting the correct result when querying the app locally from a web browser?

I can not test that at the moment, I need to get API keys for the services first. Not sure if I can get that done before our meeting though, since I also need to prepare our vacation.

why-not-try-calmer commented:
> > Okay, I think this PR is now worth considering for a merge. But before that, maybe you have a clue why the unittests for reddit and twitter fail (404) even though I am getting the correct result when querying the app locally from a web browser?
> 
> I can not test that at the moment, I need to get API keys for the services first. Not sure if I can get that done before our meeting though, since I also need to prepare our vacation.

Okay, thanks for the approval, no problem for testing, I'll just leave it as it. We'll merge later.

why-not-try-calmer committed:
- solved conflicts, rebased, updated, removed non-used
